### PR TITLE
Refactor common Navbar component

### PR DESCRIPTION
**Pull request by rorysaur**
+ Adds Kernel colors to Tailwind config of the admin app.
+ Breaks out separate `NavbarLink` component.
+ Navbar's "home" link, content/links, background color, and text color are customizable and no longer hard-coded.
+ Minor styling tweaks on the common Navbar.
+ Styling of the admin app navbar to use Kernel colors.

Caveat: The navbars inside the admin app (on Dashboard, Entities, Resources pages) don't have the Kernel styling or the links, for several reasons:
+ The links we currently have ("Learn", "Tweet", "Contribute" etc) don't make sense inside the admin app anyway. When we do want navbar links for the logged-in steward (maybe "Settings", "Sign out" etc), we can add those.
+ Currently, every page renders its own Navbar component, which is not ideal. Rather than copy/paste the Navbar content/styling so that it matches on every page, we can refactor later so that there is a single Navbar for the logged-in experience.

Default:
Screen Shot 2022-04-15 at 8 46 31 PM

Expanded:
Screen Shot 2022-04-15 at 8 46 44 PM

### CODE REVIEW

File: packages/admin/src/views/Admin.js [hunk 1]
```diff
@@ -6,13 +6,48 @@
...
-import { useServices, Navbar, FooterSmall } from '@kernel/common'
+import { useNavigate, Link } from "react-router-dom"
+import { useServices, Navbar, FooterSmall } from "@kernel/common"
 
-import bgImage from 'assets/images/admin_bg.png'
+import bgImage from "assets/images/admin_bg.png"
 
 const ADMIN_ROLE = 100 
 
+const navbarLinks = [
```
Comment (simonkernel) [inline packages/admin/src/views/Admin.js]: I think this is a good time to introduce a `App.config.json` configuration for our apps to put all these `const` I have been using across all the different files.

Let's put it next to our `App.js` @ `packages/admin/src/App.config.json`:
```
{
  "appTitle": "Kernel Steward",
  "adminRole": 100,
  "navbarLinks": [ ... ],
  ...
}
```

We can then just import as -
```
import AppConfig from 'AppConfig'
```

and use it like

```
<Navbar title={ AppConfig.appTitle } ... />
```
Let's start with that approach and just import it at the top of the file. 

As a next step, we may move to using a `Context` [0] especially once we want to support things like dark mode and other variations. Alternatively, we could also pass it down the components tree from the root. Pros and cons to both approaches for us to consider.

[0] https://reactjs.org/docs/context.html

**Review by simonkernel — CHANGES_REQUESTED** [review]
Thanks @rorysaur for submitting your first PR - you are a rockstar :star2: !

There's one main change I'd like to do while we're at it with introducing an AppConfig json otherwise things look good.

I also noticed some style differences and inconsistencies with the existing code so I'm going to introduce a linter step so we're making sure we don't spend any cycles nit picking single vs double quotes etc. No need to address this in this PR.

Let me know if you have any questions and excited to have a contributor :rocket:

**Comment by rorysaur** [comment]
@simonkernel Sounds good. I was going to propose adding a linter as well, and we can go over the set of linter rules / style guide that we want to start with. (I'm sure I also introduced inconsistencies with whitespace and semicolons.)

Re: single/double quotes, historically I've used single quotes for JS, but in these files, there is so much related to the Tailwind classes which use double quotes, that it's kind of taxing to switch between single vs. double quotes based on whether the given line is technically JS or JSX/markup.

Ultimately I don't have a strong preference as long as it's consistent.

**Comment by simonpure** [comment]
Sounds good.

Wrt linting, I'm looking at standard js [0] and also use its formatting. I did a quick test run but want to look at it in a bit more detail before committing. The alternative is configure eslint using our own rule set.

I'll keep you posted.

[0] https://standardjs.com/

**Comment by simonkernel** [comment]
fyi I think I've properly fixed the tailwind css issue in common components. The fix was a lot simpler and we just had to include the path explicitly as opposed to relying on webpack -
https://github.com/simonkernel/kernel-services/commit/292dce6eef2bc4a630471046596f27e120278005

Anyway, make sure to merge in the changes into your local branch.

**Comment by rorysaur** [comment]
@simonkernel updated, see latest commit.
Note the import statements that worked for me are like:
```
import AppConfig from '../App.config'
```

If you know how to alias it so that we can do:
```
import AppConfig from 'AppConfig'
```

Feel free to fix it after merge. Or if it's easy, you can tell me how and I'll try it.

File: packages/admin/src/views/Admin.js [hunk 2]
```diff
@@ -8,34 +8,10 @@
 
 import { useNavigate, Link } from "react-router-dom"
 import { useServices, Navbar, FooterSmall } from "@kernel/common"
+import AppConfig from "../App.config"
```
Comment (simonkernel) [inline packages/admin/src/views/Admin.js]: We're using absolute imports [0] in the web apps, so should just be

```
import AppConfig from 'App.config'
```

[0] https://create-react-app.dev/docs/importing-a-component/#absolute-imports

File: packages/admin/tailwind.config.js [hunk 3]
```diff
@@ -3,7 +3,29 @@ module.exports = {
     "./src/**/*.{js,jsx,ts,tsx}"
```
Comment (simonkernel) [inline packages/admin/tailwind.config.js]: This doesn't seem to have the changes to fix the CSS from common components I've pushed in the meantime.

**Review by simonkernel — CHANGES_REQUESTED** [review]
Thanks! 

Your import statement is correct and this was a typo on my end. Unless of course we think we should change the file name, but I say let's go with this.

This looks good besides a minor nit about absolute imports and we can always fix this later.

Before merging this into the main repo, please first fetch and rebase your fork so the latest change I pushed are reflected (ping me if you have any questions).

Thanks!

**Comment by rorysaur** [comment]
Updated and rebased.

**Review by simonkernel — APPROVED** [review]
LGTM

**Pull request merged**